pkg/txm/storage/inmemory_store.go

```diff
@@ -277,6 +300,11 @@ func (m *InMemoryStore) UpdateUnstartedTransactionWithNonce(nonce uint64) (*type
 		m.lggr.Debugf("Unstarted transactions queue is empty for address: %v", m.address)
 		return nil, nil
 	}
+	prunedTxIDs := m.pruneUnstartedTransactionsWithinDuration(pruneUnstartedTxDuration)
+	if prunedTxIDs != nil {
```

Comment:
It's safer to check like this:
```suggestion
	if len(prunedTxIDs) != 0 {
```
because depending on the implementation, a slice could be `!= nil`, but still have 0 elements. `len` method is safe to use even for nil slices.